Jep454Tests_testLinkerFfi_DownCall test_addDoubleAndFloatFromUnionWithDoubleFloat/FloatDouble AssertionError expected but found Jep454Tests_testLinkerFfi_UpCall Bus error ffi_call_unix64+0x52

https://openj9-jenkins.osuosl.org/job/Test_openjdk25_j9_sanity.functional_x86-64_mac_Nightly_testList_0/221 - [mac15-x86-1](https://openj9-jenkins.osuosl.org/computer/mac15-x86-1)
Jep454Tests_testLinkerFfi_DownCall_0
```
03:58:21  FAILED: test_addDoubleAndFloatFromUnionWithDoubleFloat
03:58:21  java.lang.AssertionError: expected [235.886] but found [216.666]
03:58:21  	at org.testng.Assert.fail(Assert.java:96)
03:58:21  	at org.testng.Assert.failNotEquals(Assert.java:776)
03:58:21  	at org.testng.Assert.assertEquals(Assert.java:489)
03:58:21  	at org.testng.Assert.assertEquals(Assert.java:502)
03:58:21  	at org.openj9.test.jep454.downcall.UnionTests.test_addDoubleAndFloatFromUnionWithDoubleFloat(UnionTests.java:2156)

03:58:21  FAILED: test_addDoubleAndFloatFromUnionWithFloatDouble
03:58:21  java.lang.AssertionError: expected [132.011] but found [113.567]
03:58:21  	at org.testng.Assert.fail(Assert.java:96)
03:58:21  	at org.testng.Assert.failNotEquals(Assert.java:776)
03:58:21  	at org.testng.Assert.assertEquals(Assert.java:489)
03:58:21  	at org.testng.Assert.assertEquals(Assert.java:502)
03:58:21  	at org.openj9.test.jep454.downcall.UnionTests.test_addDoubleAndFloatFromUnionWithFloatDouble(UnionTests.java:2137)
```

Jep454Tests_testLinkerFfi_UpCall_0
```
03:59:21  Type=Bus error vmState=0x00000000
03:59:21  J9Generic_Signal_Number=00000028 Signal_Number=0000000a Error_Value=00000000 Signal_Code=00000002
03:59:21  Handler1=0000000007F23B00 Handler2=0000000007C44540 InaccessibleAddress=000000004199C28F Sending_Process=00000000
03:59:21  RDI=000000004199C28F RSI=0000000007B0E128 RAX=0000000000000001 RBX=00007000016B9118
03:59:21  RCX=00007FE8B584FC68 RDX=0000000000000000 R8=00007000016B8B40 R9=00007FF814C7A7BD
03:59:21  R10=0000000000000000 R11=0000000011ECBE90 R12=0000000011ECBE90 R13=0000000000000809
03:59:21  R14=0000000000000000 R15=0000000000000000
03:59:21  RIP=000000004199C28F GS=0000 FS=0000 RSP=00007000016B8B78
03:59:21  RFlags=0000000000010202 CS=002B RBP=00007000016B8B80 ERR=4199C28F00000014
03:59:21  TRAPNO=000000140000000E CPU=C28F000000140000 FAULTVADDR=000000004199C28F
03:59:21  XMM0=406b154fdf3b645a (f: 3745211392.000000, d: 2.166660e+02)
03:59:21  XMM1=00007fe8b584fc48 (f: 3045391360.000000, d: 6.948413e-310)
03:59:21  XMM2=0000000000000000 (f: 0.000000, d: 0.000000e+00)
03:59:21  XMM3=0000000000000001 (f: 1.000000, d: 4.940656e-324)
03:59:21  XMM4=0000000000000000 (f: 0.000000, d: 0.000000e+00)
03:59:21  XMM5=00007000016b8ba0 (f: 23825312.000000, d: 6.084188e-310)
03:59:21  XMM6=00007000016b8ba0 (f: 23825312.000000, d: 6.084188e-310)
03:59:21  XMM7=00007fe8b6808850 (f: 3061876736.000000, d: 6.948414e-310)
03:59:21  XMM8=0000000200000002 (f: 2.000000, d: 4.243992e-314)
03:59:21  XMM9=62577a8c8cb1c40c (f: 2360460288.000000, d: 5.408188e+165)
03:59:21  XMM10=0000000000000000 (f: 0.000000, d: 0.000000e+00)
03:59:21  XMM11=ca62c1d6ca62c1d6 (f: 3395469824.000000, d: -2.193092e+50)
03:59:21  XMM12=0000000000000000 (f: 0.000000, d: 0.000000e+00)
03:59:21  XMM13=0000000000000000 (f: 0.000000, d: 0.000000e+00)
03:59:21  XMM14=0000000000000000 (f: 0.000000, d: 0.000000e+00)
03:59:21  XMM15=0000000000000000 (f: 0.000000, d: 0.000000e+00)
03:59:21  Target=2_90_20260622_241 (Mac OS X 15.3)
03:59:21  CPU=amd64 (12 logical CPUs) (0x400000000 RAM)
03:59:21  ----------- Stack Backtrace -----------
03:59:21  (0x000000004199C28F [<unknown>+0x0])
03:59:21  ffi_call_unix64+0x52 (0x000000000819B11A [libj9vm29.dylib+0x2a411a])
03:59:21  ffi_call_int+0x3c5 (0x000000000819A915 [libj9vm29.dylib+0x2a3915])
03:59:21  ffiCallWithSetJmpForUpcall+0xa5 (0x0000000007F6B4E5 [libj9vm29.dylib+0x744e5])
03:59:21  _ZN32VM_BytecodeInterpreterCompressed3runEP10J9VMThread+0x20b4b (0x0000000007FBD0DB [libj9vm29.dylib+0xc60db])
03:59:21  bytecodeLoopCompressed+0xc8 (0x0000000007F9C578 [libj9vm29.dylib+0xa5578])
```

https://openj9-jenkins.osuosl.org/job/Test_openjdk26_j9_sanity.functional_x86-64_mac_Nightly_testList_0/127 - [mac13-x86-1](https://openj9-jenkins.osuosl.org/computer/mac13-x86-1)
```
06:32:28  FAILED: test_addDoubleAndFloatFromUnionWithDoubleFloat
06:32:28  java.lang.AssertionError: expected [235.886] but found [1.0550745077382382E34]
06:32:28  	at org.testng.Assert.fail(Assert.java:96)
06:32:28  	at org.testng.Assert.failNotEquals(Assert.java:776)
06:32:28  	at org.testng.Assert.assertEquals(Assert.java:489)
06:32:28  	at org.testng.Assert.assertEquals(Assert.java:502)
06:32:28  	at org.openj9.test.jep454.downcall.UnionTests.test_addDoubleAndFloatFromUnionWithDoubleFloat(UnionTests.java:2156)

06:32:28  FAILED: test_addDoubleAndFloatFromUnionWithFloatDouble
06:32:28  java.lang.AssertionError: expected [132.011] but found [1.0550745077382382E34]
06:32:28  	at org.testng.Assert.fail(Assert.java:96)
06:32:28  	at org.testng.Assert.failNotEquals(Assert.java:776)
06:32:28  	at org.testng.Assert.assertEquals(Assert.java:489)
06:32:28  	at org.testng.Assert.assertEquals(Assert.java:502)
06:32:28  	at org.openj9.test.jep454.downcall.UnionTests.test_addDoubleAndFloatFromUnionWithFloatDouble(UnionTests.java:2137)
```

Jep454Tests_testLinkerFfi_UpCall_0

```
06:33:33  Type=Bus error vmState=0x00000000
06:33:33  J9Generic_Signal_Number=00000028 Signal_Number=0000000a Error_Value=00000000 Signal_Code=00000002
06:33:33  Handler1=0000000001FDDB00 Handler2=0000000001CE6540 InaccessibleAddress=000000004199C28F Sending_Process=00000000
06:33:33  RDI=000000004199C28F RSI=0000000001B92128 RAX=0000000000000001 RBX=000070000C8C3118
06:33:33  RCX=00007FF2E30C7260 RDX=00007FF2E30C7268 R8=000070000C8C2B40 R9=00007FF80DADD708
06:33:33  R10=0000000000000000 R11=000000000C4B9E90 R12=000000000C4B9E90 R13=0000000000000809
06:33:33  R14=0000000000000000 R15=0000000000000000
06:33:33  RIP=000000004199C28F GS=0000 FS=0000 RSP=000070000C8C2B78
06:33:33  RFlags=0000000000010202 CS=002B RBP=000070000C8C2B80 ERR=4199C28F00000014
06:33:33  TRAPNO=000000140000000E CPU=C28F000000140000 FAULTVADDR=000000004199C28F
06:33:33  XMM0=406b154fdf3b645a (f: 3745211392.000000, d: 2.166660e+02)
06:33:33  XMM1=00007ff2e30c7248 (f: 3809243648.000000, d: 6.950573e-310)
06:33:33  XMM2=0000000000000000 (f: 0.000000, d: 0.000000e+00)
06:33:33  XMM3=0000000000000001 (f: 1.000000, d: 4.940656e-324)
06:33:33  XMM4=0000000000000000 (f: 0.000000, d: 0.000000e+00)
06:33:33  XMM5=000070000c8c2ba0 (f: 210512800.000000, d: 6.084197e-310)
06:33:33  XMM6=000070000c8c2ba0 (f: 210512800.000000, d: 6.084197e-310)
06:33:33  XMM7=00007ff2e480b650 (f: 3833640448.000000, d: 6.950574e-310)
06:33:33  XMM8=0000000000000002 (f: 2.000000, d: 9.881313e-324)
06:33:33  XMM9=3547b95adb90161c (f: 3683653120.000000, d: 4.953811e-52)
06:33:33  XMM10=0000000000000000 (f: 0.000000, d: 0.000000e+00)
06:33:33  XMM11=ca62c1d6ca62c1d6 (f: 3395469824.000000, d: -2.193092e+50)
06:33:33  XMM12=0000000000000000 (f: 0.000000, d: 0.000000e+00)
06:33:33  XMM13=0000000000000000 (f: 0.000000, d: 0.000000e+00)
06:33:33  XMM14=0000000000000000 (f: 0.000000, d: 0.000000e+00)
06:33:33  XMM15=0000000000000000 (f: 0.000000, d: 0.000000e+00)
06:33:33  Target=2_90_20260622_130 (Mac OS X 13.7.3)
06:33:33  CPU=amd64 (12 logical CPUs) (0x400000000 RAM)
06:33:33  ----------- Stack Backtrace -----------
06:33:33  (0x000000004199C28F [<unknown>+0x0])
06:33:33  ffi_call_unix64+0x52 (0x00000000022550BA [libj9vm29.dylib+0x2b10ba])
06:33:33  ffi_call_int+0x3c5 (0x00000000022548B5 [libj9vm29.dylib+0x2b08b5])
06:33:33  ffiCallWithSetJmpForUpcall+0xa5 (0x00000000020254E5 [libj9vm29.dylib+0x814e5])
06:33:33  _ZN32VM_BytecodeInterpreterCompressed3runEP10J9VMThread+0x20b4b (0x000000000207715B [libj9vm29.dylib+0xd315b])
06:33:33  bytecodeLoopCompressed+0xc8 (0x00000000020565F8 [libj9vm29.dylib+0xb25f8])
```

https://openj9-jenkins.osuosl.org/job/Test_openjdk27_j9_sanity.functional_x86-64_mac_Nightly_testList_0/2 - [mac14-x86-1](https://openj9-jenkins.osuosl.org/computer/mac14-x86-1)

Build with #24190 reverted (using my branch https://github.com/nbhuiyan/openj9/tree/revert-24190) shows that the failure is present even without #24190. [Internal build link](https://hyc-runtimes-jenkins.swg-devops.com/view/OpenJ9%20-%20Personal/job/Pipeline-Build-Test-Personal/34167/).
```
16:00:17  There are 446 test targets in testList
16:00:17  "/Users/jenkins/workspace/Test_openjdk25_j9_sanity.functional_x86-64_mac_Personal_testList_3/jdkbinary/j2sdk-image/bin/java" -version
16:00:17  openjdk version "25.0.4-internal" 2026-07-21
16:00:17  OpenJDK Runtime Environment (build 25.0.4-internal-adhoc.jenkins.BuildJDK25x86-64macPersonal)
16:00:17  Eclipse OpenJ9 VM (build 25.0.4-internal-adhoc.jenkins.BuildJDK25x86-64macPersonal-revert-24190-96675a3e0fe, JRE 25 Mac OS X amd64-64-Bit Compressed References 20260623_88 (JIT enabled, AOT enabled)
16:00:17  OpenJ9   - 96675a3e0fe
16:00:17  OMR      - a39ef2b2cfd
16:00:17  JCL      - 621fa6c0c01 based on jdk-25.0.4+5)
...
16:57:38  FAILED: test_addDoubleAndFloatFromUnionWithDoubleFloat
16:57:38  java.lang.AssertionError: expected [235.886] but found [216.666]
16:57:38  	at org.testng.Assert.fail(Assert.java:96)
16:57:38  	at org.testng.Assert.failNotEquals(Assert.java:776)
16:57:38  	at org.testng.Assert.assertEquals(Assert.java:489)
16:57:38  	at org.testng.Assert.assertEquals(Assert.java:502)
16:57:38  	at org.openj9.test.jep454.downcall.UnionTests.test_addDoubleAndFloatFromUnionWithDoubleFloat(UnionTests.java:2156)
```

This rules out #24190 as the cause.

this also still fails with https://github.com/eclipse-openj9/openj9/pull/23540 reverted 
https://hyc-runtimes-jenkins.swg-devops.com/view/OpenJ9%20-%20Personal/job/Pipeline-Build-Test-Personal/34169/

```
not ok 334 - Jep454Tests_testLinkerFfi_DownCall_0

 FAILED: test_addDoubleAndFloatFromUnionWithDoubleFloat
        java.lang.AssertionError: expected [235.886] but found [-1.960789815843523E22]
        	at org.testng.Assert.fail(Assert.java:96)
        	at org.testng.Assert.failNotEquals(Assert.java:776)
        	at org.testng.Assert.assertEquals(Assert.java:489)
        	at org.testng.Assert.assertEquals(Assert.java:502)
        	at org.openj9.test.jep454.downcall.UnionTests.test_addDoubleAndFloatFromUnionWithDoubleFloat(UnionTests.java:2156)
        	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
        	at java.base/java.lang.reflect.Method.invoke(Method.java:571)
        	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
        	at org.testng.internal.Invoker.invokeMethod(Invoker.java:580)
        	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:716)
        	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:988)
        	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
        	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:109)
        	at org.testng.TestRunner.privateRun(TestRunner.java:648)
        	at org.testng.TestRunner.run(TestRunner.java:505)
        	at org.testng.SuiteRunner.runTest(SuiteRunner.java:455)
        	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:450)
        	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:415)
        	at org.testng.SuiteRunner.run(SuiteRunner.java:364)
        	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
        	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:84)
        	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1208)
        	at org.testng.TestNG.runSuitesLocally(TestNG.java:1137)
        	at org.testng.TestNG.runSuites(TestNG.java:1049)
        	at org.testng.TestNG.run(TestNG.java:1017)
        	at org.testng.TestNG.privateMain(TestNG.java:1354)
        	at org.testng.TestNG.main(TestNG.java:1323)
        
        FAILED: test_addDoubleAndFloatFromUnionWithFloatDouble
        java.lang.AssertionError: expected [132.011] but found [-1.960789815843523E22]
        	at org.testng.Assert.fail(Assert.java:96)
        	at org.testng.Assert.failNotEquals(Assert.java:776)
        	at org.testng.Assert.assertEquals(Assert.java:489)
        	at org.testng.Assert.assertEquals(Assert.java:502)
        	at org.openj9.test.jep454.downcall.UnionTests.test_addDoubleAndFloatFromUnionWithFloatDouble(UnionTests.java:2137)
        	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
        	at java.base/java.lang.reflect.Method.invoke(Method.java:571)
        	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
        	at org.testng.internal.Invoker.invokeMethod(Invoker.java:580)
        	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:716)
        	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:988)
        	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
        	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:109)
        	at org.testng.TestRunner.privateRun(TestRunner.java:648)
        	at org.testng.TestRunner.run(TestRunner.java:505)
        	at org.testng.SuiteRunner.runTest(SuiteRunner.java:455)
        	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:450)
        	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:415)
        	at org.testng.SuiteRunner.run(SuiteRunner.java:364)
        	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
        	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:84)
        	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1208)
        	at org.testng.TestNG.runSuitesLocally(TestNG.java:1137)
        	at org.testng.TestNG.runSuites(TestNG.java:1049)
        	at org.testng.TestNG.run(TestNG.java:1017)
        	at org.testng.TestNG.privateMain(TestNG.java:1354)
        	at org.testng.TestNG.main(TestNG.java:1323)
```
and 

```
not ok 343 - Jep454Tests_testLinkerFfi_UpCall_0

Unhandled exception
        Type=Bus error vmState=0x00000000
        J9Generic_Signal_Number=00000028 Signal_Number=0000000a Error_Value=00000000 Signal_Code=00000002
        Handler1=000000000805DB00 Handler2=0000000007D7E4E0 InaccessibleAddress=000000004199C28F Sending_Process=00000000
        RDI=000000004199C28F RSI=0000000007C7B128 RAX=0000000000000001 RBX=0000700007108118
        RCX=00007FCA5C8AB660 RDX=00000000FFFFFF00 R8=0000700007107B40 R9=00007FF81E0F1A32
        R10=0000000000000000 R11=000000000840BE90 R12=000000000840BE90 R13=0000000000000809
        R14=0000000000000000 R15=0000000000000000
        RIP=000000004199C28F GS=0000 FS=0000 RSP=0000700007107B78
        RFlags=0000000000010202 CS=002B RBP=0000700007107B80 ERR=4199C28F00000014
        TRAPNO=000000140000000E CPU=C28F000000140000 FAULTVADDR=000000004199C28F
        XMM0=406b154fdf3b645a (f: 3745211392.000000, d: 2.166660e+02)
        XMM1=00007fca5c8ab648 (f: 1552594560.000000, d: 6.941974e-310)
        XMM2=0000000000000000 (f: 0.000000, d: 0.000000e+00)
        XMM3=0000000000000001 (f: 1.000000, d: 4.940656e-324)
        XMM4=0000000000000000 (f: 0.000000, d: 0.000000e+00)
        XMM5=0000700007107ba0 (f: 118520736.000000, d: 6.084192e-310)
        XMM6=0000700007107ba0 (f: 118520736.000000, d: 6.084192e-310)
        XMM7=00007fca5c811450 (f: 1551963264.000000, d: 6.941974e-310)
        XMM8=00007fca60caaa78 (f: 1623894656.000000, d: 6.941977e-310)
        XMM9=d4b8b659a379500a (f: 2742636544.000000, d: -1.351300e+100)
        XMM10=0000000000000000 (f: 0.000000, d: 0.000000e+00)
        XMM11=ca62c1d6ca62c1d6 (f: 3395469824.000000, d: -2.193092e+50)
        XMM12=0000000000000000 (f: 0.000000, d: 0.000000e+00)
        XMM13=0000000000000000 (f: 0.000000, d: 0.000000e+00)
        XMM14=0000000000000000 (f: 0.000000, d: 0.000000e+00)
        XMM15=0000000000000000 (f: 0.000000, d: 0.000000e+00)
        Target=2_90_20260623_90 (Mac OS X 12.7.6)
        CPU=amd64 (4 logical CPUs) (0x200000000 RAM)
        ----------- Stack Backtrace -----------
        (0x000000004199C28F [<unknown>+0x0])
        ffi_call_unix64+0x52 (0x00000000082D511A [libj9vm29.dylib+0x2a411a])
        ffi_call_int+0x3c5 (0x00000000082D4915 [libj9vm29.dylib+0x2a3915])
        ffiCallWithSetJmpForUpcall+0xa5 (0x00000000080A58E5 [libj9vm29.dylib+0x748e5])
        _ZN32VM_BytecodeInterpreterCompressed3runEP10J9VMThread+0x20b4b (0x00000000080F74DB [libj9vm29.dylib+0xc64db])
        bytecodeLoopCompressed+0xc8 (0x00000000080D6978 [libj9vm29.dylib+0xa5978])
```

--- a/debugtools/DDR_Autoblob/src/com/ibm/j9ddr/autoblob/GetNativeDirectory.java
+++ b/debugtools/DDR_Autoblob/src/com/ibm/j9ddr/autoblob/GetNativeDirectory.java
@@ -80,7 +80,7 @@ public class GetNativeDirectory
 		}
 
 		// if the current system is AMD64 use x86 as the osArch
-		else if (osArch.equals("amd64")) {
+		else if (osArch.equals("amd64") || osArch.equals("x86_64")) {
 			osArch = "x86";
 		}
 

--- a/jcl/src/java.base/share/classes/openj9/internal/foreign/abi/LayoutStrPreprocessor.java
+++ b/jcl/src/java.base/share/classes/openj9/internal/foreign/abi/LayoutStrPreprocessor.java
@@ -56,7 +56,7 @@ import jdk.incubator.foreign.ValueLayout;
 final class LayoutStrPreprocessor {
 	private static final String osName = System.getProperty("os.name");
 	private static final String arch = System.getProperty("os.arch");
-	private static final boolean isX86_64 = arch.equals("amd64");
+	private static final boolean isX86_64 = arch.equals("amd64") || arch.equals("x86_64");
 	private static final boolean isLinuxS390x = arch.equals("s390x") && osName.equals("Linux");
 
 	/*[IF JAVA_SPEC_VERSION == 17]*/

--- a/test/functional/Java17andUp/src_170/org/openj9/test/jep389/valist/DowncallTests.java
+++ b/test/functional/Java17andUp/src_170/org/openj9/test/jep389/valist/DowncallTests.java
@@ -57,7 +57,7 @@ import jdk.incubator.foreign.ValueLayout;
 public class DowncallTests {
 	private static final String osName = System.getProperty("os.name");
 	private static final String arch = System.getProperty("os.arch");
-	private static final boolean isX64 = arch.equals("amd64");
+	private static final boolean isX64 = arch.equals("amd64") || arch.equals("x86_64");
 	private static final boolean isWinX64 = osName.startsWith("Windows") && isX64;
 	private static final boolean isLinuxAarch64 = osName.equals("Linux") && arch.equals("aarch64");
 	/* The padding of struct is not required on Power in terms of VaList */

--- a/test/functional/Java17andUp/src_170/org/openj9/test/jep389/valist/UpcallTests.java
+++ b/test/functional/Java17andUp/src_170/org/openj9/test/jep389/valist/UpcallTests.java
@@ -57,7 +57,7 @@ import jdk.incubator.foreign.ValueLayout;
 public class UpcallTests {
 	private static final String osName = System.getProperty("os.name");
 	private static final String arch = System.getProperty("os.arch");
-	private static final boolean isX64 = arch.equals("amd64");
+	private static final boolean isX64 = arch.equals("amd64") || arch.equals("x86_64");
 	private static final boolean isWinX64 = osName.startsWith("Windows") && isX64;
 	private static final boolean isLinuxAarch64 = osName.equals("Linux") && arch.equals("aarch64");
 	/* The padding of struct is not required on Power in terms of VaList */

--- a/test/functional/Java8andUp/src/org/openj9/test/contendedfields/ContendedFieldsTests.java
+++ b/test/functional/Java8andUp/src/org/openj9/test/contendedfields/ContendedFieldsTests.java
@@ -59,7 +59,7 @@ public class ContendedFieldsTests {
 			CACHE_LINE_SIZE = 128;
 		} else if (osArch.startsWith("s390")) {
 			CACHE_LINE_SIZE = 256;
-		} else if (osArch.equals("amd64") || osArch.equals("x86")) {
+		} else if (osArch.equals("amd64") || osArch.equals("x86") || osArch.equals("x86_64")) {
 			CACHE_LINE_SIZE = 64;
 		} else if (osArch.equals("aarch64")) {
 			String osName = System.getProperty("os.name");

--- a/test/functional/VM_Test/src/j9vm/runner/Runner.java
+++ b/test/functional/VM_Test/src/j9vm/runner/Runner.java
@@ -73,7 +73,7 @@ public class Runner {
 				return PPC;
 			} else if (archSpec.startsWith("s390")) {
 				return S390X;
-			} else if (archSpec.equals("amd64") || archSpec.equals("x86")) {
+			} else if (archSpec.equals("amd64") || archSpec.equals("x86") || archSpec.equals("x86_64")) {
 				return X86;
 			} else {
 				System.out.println("Runner couldn't determine underlying architecture. Got OS Arch:" + archSpec);

